### PR TITLE
Use file md5sum as the session key

### PR DESCRIPTION
**Pull request by ptpt**
- Use file md5sum as the upload session key
- Do not caculate the sequence md5sum

### CODE REVIEW

**Comment by GITNE** [comment]
Same goes here for MD5 and elsewhere in the code. You are strongly recommended to migrate to SHA1 or later. Due to the large amount of data handled every day, you are going to see collisions. Users are going to experience bogus errors and everybody involved will not even have clue about what is going on.

**Comment by ptpt** [comment]
Hey @GITNE In our case the md5sum is calculated at sequence level per user. Assume a user has uploaded 1M sequences, then the collision probability of the next upload is around 1,000,000 * 1.47*10-29 = 1.47e-22, which is quite acceptable IMO.

**Comment by GITNE** [comment]
> 1,000,000 * 1.47*10-29 = 1.47e-22, which is quite acceptable IMO.

Maybe, maybe not, in an ideal case which MD5 is far from. Latest analysis suggests that MD5’s collision resistance is down to 2<sup>18</sup> on a single 512 bit block. In other words, the problem with MD5 is that the probability of collisions greatly increases especially on inputs that share identical (512 bit) blocks of data on 64 octet boundaries. Hence, JPEGs with largely identical TIFF headers are a great example of inputs which are extremely prone to this issue.

Long story short: Just do not use MD5, especially in new code, regardless for what purpose or reason. Just don’t.
